src/main/java/gregtech/client/renderer/texture/custom/QuantumStorageRenderer.java

```diff
@@ -0,0 +1,147 @@
+import codechicken.lib.vec.Transformation;
+import gregtech.api.GTValues;
+import gregtech.api.gui.resources.TextTexture;
+import gregtech.api.util.TextFormattingUtil;
+import gregtech.client.renderer.texture.Textures;
+import gregtech.client.utils.RenderUtil;
+import gregtech.common.metatileentities.storage.MetaTileEntityQuantumChest;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.RenderItem;
+import net.minecraft.client.renderer.block.model.ItemCameraTransforms;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class QuantumStorageRenderer implements TextureUtils.IIconRegister {
+    private static final Cuboid6 upBox = new Cuboid6(0 / 16.0, 14 / 16.0, 0 / 16.0, 16 / 16.0, 16 / 16.0, 16 / 16.0);
+    private static final Cuboid6 downBox = new Cuboid6(0 / 16.0, 0 / 16.0, 0 / 16.0, 16 / 16.0, 2 / 16.0, 16 / 16.0);
+    private static final Cuboid6 westBox = new Cuboid6(0 / 16.0, 0 / 16.0, 0 / 16.0, 2 / 16.0, 16 / 16.0, 16 / 16.0);
+    private static final Cuboid6 eastBox = new Cuboid6(14 / 16.0, 0 / 16.0, 0 / 16.0, 16 / 16.0, 16 / 16.0, 16 / 16.0);
+    private static final Cuboid6 southBox = new Cuboid6(0 / 16.0, 0 / 16.0, 14 / 16.0, 16 / 16.0, 16 / 16.0, 16 / 16.0);
+    private static final Cuboid6 northBox = new Cuboid6(0 / 16.0, 0 / 16.0, 0 / 16.0, 16 / 16.0, 16 / 16.0, 2 / 16.0);
+    private static final Cuboid6 glassBox = new Cuboid6(1 / 16.0, 1 / 16.0, 1 / 16.0, 15 / 16.0, 15 / 16.0, 15 / 16.0);
+    private static final Cuboid6 fluidBox = new Cuboid6(2 / 16.0, 2 / 16.0, 2 / 16.0, 14 / 16.0, 14 / 16.0, 14 / 16.0);
+
+
+    private static Map<EnumFacing, Cuboid6> boxFacingMap = new HashMap<>();
+
+    @SideOnly(Side.CLIENT)
+    private TextureAtlasSprite[] textures;
+
+    private int tier;
+
+    private static Transformation[] rotations = {new Rotation(90, 0, 0, 1), new Rotation(-90, 0, 0, 1), Rotation.quarterRotations[1], Rotation.quarterRotations[2], Rotation.quarterRotations[3], Rotation.quarterRotations[0]};
+
+    static {
+        boxFacingMap.put(EnumFacing.UP, upBox);
+        boxFacingMap.put(EnumFacing.DOWN, downBox);
+        boxFacingMap.put(EnumFacing.WEST, westBox);
+        boxFacingMap.put(EnumFacing.EAST, eastBox);
+        boxFacingMap.put(EnumFacing.SOUTH, southBox);
+        boxFacingMap.put(EnumFacing.NORTH, northBox);
+    }
+
+    public QuantumStorageRenderer(int tier) {
```

Comment:
This class can be made into a singleton with a single instance, which does not store a specific tier for the machine. See my other comments about storing the two textures in the array, and in renderMachine.